File: 351004/Beliaev/restapp/publisher/src/main/java/com/example/publisher/controller/AuthController.java
package com.example.publisher.controller;

import com.example.publisher.dto.request.AuthRequest;
import com.example.publisher.security.JwtUtils;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/v2.0")
@RequiredArgsConstructor
public class AuthController {

    private final AuthenticationManager authenticationManager;
    private final JwtUtils jwtUtils;

    @PostMapping("/login")
    public Map<String, String> authenticateUser(@RequestBody AuthRequest loginRequest) {
        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(loginRequest.getLogin(), loginRequest.getPassword()));

        SecurityContextHolder.getContext().setAuthentication(authentication);
        UserDetails userDetails = (UserDetails) authentication.getPrincipal();

        String role = userDetails.getAuthorities().stream()
                .map(GrantedAuthority::getAuthority)
                .map(r -> r.replace("ROLE_", ""))
                .findFirst().orElse("CUSTOMER");

        String jwt = jwtUtils.generateJwtToken(userDetails.getUsername(), role);

        return Map.of(
                "access_token", jwt,
                "token_type", "Bearer"
        );
    }
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/controller/AuthController.kt
package com.example.entitiesapp.controller

import com.example.entitiesapp.util.JwtUtils
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v2.0")
class AuthController(
    private val authenticationManager: AuthenticationManager,
    private val jwtUtils: JwtUtils
) {
    @PostMapping("/login")
    fun login(@RequestBody authRequest: Map<String, String>): Map<String, String> {
        val login = authRequest["login"] ?: ""
        val password = authRequest["password"] ?: ""

        val auth = authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(login, password)
        )

        val role = auth.authorities.first().authority
        val token = jwtUtils.generateToken(login, role)

        return mapOf("access_token" to token)
    }
}